ncm-nss/src/main/perl/nss.pm

```diff
@@ -95,28 +95,28 @@ sub Configure {
 
     my @new = ( "# Generated by ncm-nss\n\n" );
     push(@new, map { "$_: $databases->{$_}\n" } sort keys %$databases);
-    my $contents = join("", @new);
-    # cannot use CAF::FileWriter because close() always returns false in
-    # noaction mode so cannot show when nscd would be restarted
-    my $result = LC::Check::file("/etc/nsswitch.conf",
-                                 backup => ".OLD",
-                                 contents => $contents,
-                                 owner => "root",
-                                 group => "root",
-                                 mode => "0444",
-                                );
+    my $fh = CAF::FileWriter->open("/etc/nsswitch.conf",
+                                   backup => ".OLD",
+                                   owner => "root",
+                                   group => "root",
+                                   mode => 0444,
+                                   log => $self,
+        );
+    print $fh join("", @new);
+    my $result = $fh->close();
+
     if ($result) {
-        $self->log("updated /etc/nsswitch.conf");
+        $self->info("updated /etc/nsswitch.conf") unless $NoAction;
```

Comment:
wrt `NoAction`, in the old components code, we extend the `NCM::Component` namespace by diretc manipulation of `ISA`, and the `$NoAction` variable is the one set here: https://github.com/quattor/ncm-ncd/blob/master/src/main/perl/Component.pm#L15
if you would switch to the `use parent qw(NCM::Component)`, this would fail, and you would have to use `$CAF::Object::NoAction`. as the `NoAction` is read most of the time, this is ok.